File: 品味Java的21種設計模式/dp_java/facade/example3/ConfigManager.java
package cn.javass.dp.facade.example3;
/**
 * 示意配置管理，就是负责读取配置文件，
 * 并把配置文件的内容设置到配置Model中去，是个单例
 */
public class ConfigManager {
    private static ConfigManager manager = null;
    private static ConfigModel cm = null;
    private ConfigManager() {

    }
    public static ConfigManager getInstance() {
        if (manager == null) {
            manager = new ConfigManager();
            cm = new ConfigModel();
            //读取配置文件，把值设置到ConfigModel中去
        }

        return manager;
    }
    /**
     * 获取配置的数据
     * @return 配置的数据
     */
    public ConfigModel getConfigData() {
        return cm;
    }
}

File: 品味Java的21種設計模式/dp_cpp/facade/example1/ConfigManager.cpp
#include "ConfigManager.h"

namespace cn {
namespace javass {
namespace dp {
namespace facade {
namespace example1 {

ConfigManager::ConfigManager() {

}

ConfigManager *ConfigManager::getInstance() {
    if (manager == 0) {
        manager = new ConfigManager();
        cm = new ConfigModel();
        //读取配置文件，把值设置到ConfigModel中去
    }

    return manager;
}

ConfigModel *ConfigManager::getConfigData() {
    return cm;
}
}
}
}
}
}
